jit-compiler/src/compiler.rs

```diff
@@ -111,6 +134,18 @@ impl Add for ibig::IBig {
     fn add(a: Self, b: Self) -> Self { a + b }
 }
 
+impl<T: Clone> Add for PilVec<T> {
+    fn add(a: Self, b: Self) -> Self {
+        // TODO for a regular "push" or array::map this is very slow.
+        // We could optimize this by sharing a larger backing vector
+        // across prefix instances, allowing to extend the backing vector if
+        // our view is the full vector.
```

Comment:
Maybe it should implement "push" and "extend", but then it would have to be Arc<RwLock<Vec<T>>>.

Reply:
Yeah, we can do that later, after we checked what the performance bottlenecks actually are.